apps/system/js/captive_portal.js

```diff
@@ -74,12 +76,13 @@ var CaptivePortal = {
       }
     };
 
-    this.notification = new Notification('', options);
-    this.notification.addEventListener('click',
-      this.captiveNotification_onClick);
-    this.notification.addEventListener('close', (function() {
-      this.notification = null;
-    }).bind(this));
+    NotificationHelper.send('', options).then(notification => {
+      notification.addEventListener('click',
+        this.captiveNotification_onClick(notification));
+      notification.addEventListener('close', () => {
+        notification = null;
+      });
+    });
   },
 
   dismissNotification: function dismissNotification(id) {
```

Comment:
There are a bunch of calls to `this.notification` and I'm not sure how to resolve those in the `dismissNotification` function. The old code assigned the new Notification to `this.notification`, but that doesn't seem to work with the `NotificationHelper`